### PR TITLE
fuzz-config: add config to support fuzzing test

### PR DESCRIPTION
**Pull request by libinyang**
kernel config for fuzzing test.

Signed-off-by: Libin Yang <libin.yang@intel.com>

### CODE REVIEW

File: fuzz-defconfig [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# code coverage collection
+CONFIG_KCOV=y
+CONFIG_KCOV_INSTRUMENT_ALL=y
+CONFIG_DEBUG_FS=y
```
Comment (plbossart) [inline fuzz-defconfig:4]: already set by default.

File: fuzz-defconfig [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# show code coverage in web interface
+CONFIG_DEBUG_INFO=y
+
+# namespace sandbox
+CONFIG_NAMESPACES=y
```
Comment (plbossart) [inline fuzz-defconfig:10]: already set by default

File: fuzz-defconfig [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+CONFIG_USER_NS=y
```
Comment (plbossart) [inline fuzz-defconfig:11]: already set by default

File: fuzz-defconfig [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+CONFIG_UTS_NS=y
```
Comment (plbossart) [inline fuzz-defconfig:12]: already set by default

File: fuzz-defconfig [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+CONFIG_IPC_NS=y
+CONFIG_PID_NS=y
+CONFIG_NET_NS=y
```
Comment (plbossart) [inline fuzz-defconfig:15]: all above already set

File: fuzz-defconfig [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+# KASAN for use-after-free and out-of-bounds detection
+CONFIG_KASAN=y
+CONFIG_KASAN_INLINE=y
+
+# Any other debugging configs
+CONFIG_LOCKDEP=y
+CONFIG_PROVE_LOCKING=y
+CONFIG_DEBUG_ATOMIC_SLEEP=y
+CONFIG_PROVE_RCU=y
+CONFIG_DEBUG_VM=y
```
Comment (plbossart) [inline fuzz-defconfig:26]: we already have the memory-debug defconfig. Do we need to redefine this here?

File: fuzz-defconfig [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+# Disable the following configs
+# CONFIG_RANDOMIZE_BASE is not set
+
+# Increase RCU stall timeout to reduce false positive rate
+CONFIG_RCU_CPU_STALL_TIMEOUT=60
+
+# PSTORE
+CONFIG_EFI_VARS_PSTORE=m
+CONFIG_PSTORE=m
```
Comment (plbossart) [inline fuzz-defconfig:36]: move one line up, this gates all the others below.

File: fuzz-defconfig [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+CONFIG_PSTORE_CONSOLE=y
+CONFIG_PSTORE_PMSG=y
+CONFIG_PSTORE_RAM=m
+CONFIG_PSTORE_ZONE=m
+CONFIG_PSTORE_BLK=m
+CONFIG_PSTORE_BLK_BLKDEV=""
+CONFIG_PSTORE_BLK_KMSG_SIZE=64
+CONFIG_PSTORE_BLK_MAX_REASON=2
+CONFIG_PSTORE_BLK_PMSG_SIZE=64
+CONFIG_PSTORE_BLK_CONSOLE_SIZE=64
```
Comment (plbossart) [inline fuzz-defconfig:51]: missing end of line.

**Review by plbossart — CHANGES_REQUESTED** [review]
Thanks @libinyang 

No objection to adding this fragment, but I'd like to minimize it what's not already defined by the defconfig+base. Can you try and optimize a bit?

Also the memory debug is already defined, maybe with too many options. Maybe we can split in memory-debug-base and memory-debug-extensive fragments.

**Comment by libinyang** [comment]
@plbossart Thanks for reviewing. I have removed all the duplicated definitions and suggest reusing other defconfigs in the README. For the memory-debug-defconfig, I think it should be OK. Fuzz config is copied from syzkaller suggestion of minimum configuration. And I believe applying all configs in memory-debug-defconfig is acceptable for fuzzing test.

**Pull request merged**